expression/distsql_builtin.go

```diff
@@ -287,119 +321,185 @@ func getSignatureByPB(ctx sessionctx.Context, sigCode tipb.ScalarFuncSig, tp *ti
 	case tipb.ScalarFuncSig_IntIsNull:
 		f = &builtinIntIsNullSig{base}
-
-	case tipb.ScalarFuncSig_CoalesceDecimal:
-		f = &builtinCoalesceDecimalSig{base}
-	case tipb.ScalarFuncSig_CoalesceDuration:
-		f = &builtinCoalesceDurationSig{base}
-	case tipb.ScalarFuncSig_CoalesceReal:
-		f = &builtinCoalesceRealSig{base}
-	case tipb.ScalarFuncSig_CoalesceTime:
-		f = &builtinCoalesceTimeSig{base}
-	case tipb.ScalarFuncSig_CoalesceString:
-		f = &builtinCoalesceStringSig{base}
-	case tipb.ScalarFuncSig_CoalesceInt:
-		f = &builtinCoalesceIntSig{base}
-
-	case tipb.ScalarFuncSig_CaseWhenJson:
-		f = &builtinCaseWhenJSONSig{base}
-	case tipb.ScalarFuncSig_CaseWhenDecimal:
-		f = &builtinCaseWhenDecimalSig{base}
-	case tipb.ScalarFuncSig_CaseWhenDuration:
-		f = &builtinCaseWhenDurationSig{base}
-	case tipb.ScalarFuncSig_CaseWhenReal:
-		f = &builtinCaseWhenRealSig{base}
-	case tipb.ScalarFuncSig_CaseWhenTime:
-		f = &builtinCaseWhenTimeSig{base}
-	case tipb.ScalarFuncSig_CaseWhenString:
-		f = &builtinCaseWhenStringSig{base}
-	case tipb.ScalarFuncSig_CaseWhenInt:
-		f = &builtinCaseWhenIntSig{base}
-
-	case tipb.ScalarFuncSig_IntIsFalse:
-		f = &builtinIntIsFalseSig{base, false}
-	case tipb.ScalarFuncSig_RealIsFalse:
-		f = &builtinRealIsFalseSig{base, false}
-	case tipb.ScalarFuncSig_DecimalIsFalse:
-		f = &builtinDecimalIsFalseSig{base, false}
+	//case tipb.ScalarFuncSig_JsonIsNull:
+	case tipb.ScalarFuncSig_BitAndSig:
+		f = &builtinBitAndSig{base}
+	case tipb.ScalarFuncSig_BitOrSig:
+		f = &builtinBitOrSig{base}
+	case tipb.ScalarFuncSig_BitXorSig:
+		f = &builtinBitXorSig{base}
+	case tipb.ScalarFuncSig_BitNegSig:
+		f = &builtinBitNegSig{base}
 	case tipb.ScalarFuncSig_IntIsTrue:
 		f = &builtinIntIsTrueSig{base, false}
 	case tipb.ScalarFuncSig_RealIsTrue:
 		f = &builtinRealIsTrueSig{base, false}
 	case tipb.ScalarFuncSig_DecimalIsTrue:
 		f = &builtinDecimalIsTrueSig{base, false}
-	case tipb.ScalarFuncSig_ModInt:
-		f = &builtinArithmeticModIntSig{base}
-	case tipb.ScalarFuncSig_ModReal:
-		f = &builtinArithmeticModRealSig{base}
-	case tipb.ScalarFuncSig_ModDecimal:
-		f = &builtinArithmeticModDecimalSig{base}
-
-	case tipb.ScalarFuncSig_IfNullReal:
-		f = &builtinIfNullRealSig{base}
+	case tipb.ScalarFuncSig_IntIsFalse:
+		f = &builtinIntIsFalseSig{base, false}
+	case tipb.ScalarFuncSig_RealIsFalse:
+		f = &builtinRealIsFalseSig{base, false}
+	case tipb.ScalarFuncSig_DecimalIsFalse:
+		f = &builtinDecimalIsFalseSig{base, false}
+	case tipb.ScalarFuncSig_LeftShift:
+		f = &builtinLeftShiftSig{base}
+	case tipb.ScalarFuncSig_RightShift:
+		f = &builtinRightShiftSig{base}
+	case tipb.ScalarFuncSig_BitCount:
+		f = &builtinBitCountSig{base}
+	case tipb.ScalarFuncSig_GetParamString:
+		f = &builtinGetParamStringSig{base}
+	case tipb.ScalarFuncSig_GetVar:
+		f = &builtinGetVarSig{base}
+	//case tipb.ScalarFuncSig_RowSig:
+	case tipb.ScalarFuncSig_SetVar:
+		f = &builtinSetVarSig{base}
+	//case tipb.ScalarFuncSig_ValuesDecimal:
```

Comment:
The comment case means?

Reply:
Means they are either not implemented by TiDB (so that there is no `builtinXxxSig`), or they are implemented but require additional parameters (like `{base, xx, yy}`) that I don't know how to fill for now.